Tsi m85 v1

gkethamallax opened the pull request:
Hi Team,

I updated the board configuration from **ek_tsi** to **tsi** and resolved compliance errors in the SoC and board configurations.
Please use the updated build command for m85: **west build -p always -b tsi/skyp/m85 samples/hello_world**

Build command for adsp: "west build -p always -b tsi/mimx8ml8/adsp samples/hello_world/"

gkethamallax committed:
TSI: board name changed

-ek_tsi to tsi board name changed
-comilance erros removed from some files

Signed-off-by: Ganesh Kethamalla <gkethamallax@tsavoritesi.com>

gkethamallax committed:
TSI Board: compliance errors removed and and board variant name changed

-board variant name changed to m85 to cortex-m85
-white spaces and extra lines removed

Signed-off-by: Ganesh Kethamalla <gkethamallax@tsavoritesi.com>

gkethamallax committed:
TSI Board: old board file removed from directory

-created tsi new directory, instead of using ek_tsi. that's removed old one.

Signed-off-by: Ganesh Kethamalla <gkethamallax@tsavoritesi.com>

Commit:
added new cpu configuration adsp

Commit:
Modified TXE address for DV environment testing

atrivedi-tsavoritesi commented on boards/tsi/tsi/tsi_mimx8ml8_adsp.dts:
```diff
@@ -14,11 +14,17 @@
+	sram_txe: memory@60040000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x60040000 DT_SIZE_K(512)>;
```
We can call this as tsi_txe.dts, instead of calling it tsi_mimx8ml8_adsp ? is that doable ? I can give you pointers on how to add it. Also this current sram_txe we were thinking will be sram1 so we can call it memory@60640000 and change reg to 0x60640000

atrivedi-tsavoritesi commented on soc/nxp/imx/imx8m/adsp/memory.h:
```diff
@@ -78,20 +86,27 @@
-#define DRAM0_BASE	0x3B6E8000
-#define DRAM0_SIZE	0x8000
+/*#define DRAM0_BASE	0x3B6E8000*/
+#define DRAM0_BASE  	0x70800000
```
you may want to address the lint warning.

atrivedi-tsavoritesi commented on soc/nxp/imx/imx8m/adsp/memory.h:
```diff
@@ -78,20 +86,27 @@
-#define DRAM1_BASE	0x3B6F0000
-#define DRAM1_SIZE	0x8000
+/*#define DRAM1_BASE	0x3B6F0000*/
+#define DRAM1_BASE  	0x80608000
```
Same here, by the way was your intent to make this 0x8060 or was it 0x6060 for DRAM1? Never mind this should be ok for now.

atrivedi-tsavoritesi commented on soc/nxp/imx/imx8m/adsp/memory.h:
```diff
@@ -78,20 +86,27 @@
-#define SDRAM0_BASE	0x92400000
-#define SDRAM0_SIZE	0x800000
+/*#define SDRAM0_BASE	0x92400000*/
+#define SDRAM0_BASE 	0x60601000
```
Same fix the lint issues.

atrivedi-tsavoritesi commented on boards/tsi/tsi/board.yml:
```diff
@@ -0,0 +1,10 @@
+board:
+  name: tsi
+  vendor: Tsavorite
```
Is there a place we can define Tsavorite as this is flagging an error.

atrivedi-tsavoritesi commented on boards/tsi/tsi/board.yml:
```diff
@@ -0,0 +1,10 @@
+  - name: skyp
+    variants:
+    - name: m85
+      cpucluster: m85
```
Was this giving two m85 board types right ? is it fixed ?

gkethamallax replied:
If we are using only the SoC without variants, it is possible to run with one M85. but, if we want to use the ADSP, we need to provide the necessary configurations with variants.

atrivedi-tsavoritesi commented on boards/tsi/tsi/Kconfig.tsi:
```diff
@@ -2,4 +2,6 @@
 
 config BOARD_TSI
-	select SOC_SKYP
+	select SOC_MIMX8ML8_ADSP if BOARD_TSI_MIMX8ML8_ADSP
```
I know you mentioned that we had to use MIIMX8ML8 requirement somewhere but since these are new configs we can define these right ? I had defined something similar in the earlier PRs in main branch you can look it up.

atrivedi-tsavoritesi replied:
resolving this comment as I made changes to address this, the new target is "tsi/skyp/txe" now for TXE.

atrivedi-tsavoritesi commented on boards/tsi/tsi/board.yml:
```diff
@@ -1,10 +1,9 @@
       cpucluster: m85
-    # - name: xtensa
-    #   cpucluster: xtensa
+  - name: mimx8ml8
```
If you remove this mim8ml8 and just make it txe, in this file board/tsi/tsi/board.yml file you can remove the requirement of this being called imx8ml8 so you can just call it txe without adsp and 8imx8ml8 extension requirement.

atrivedi-tsavoritesi replied:
resolving this comment as I made changes to address this, the new target is "tsi/skyp/txe" now for TXE.

atrivedi-tsavoritesi committed:
@FIR-13: Create a new target "west build -p always -b tsi/skyp/txe samples/hello_world/"
and remove the old target "west build -p always -b tsi/mimx8ml8/adsp samples/hello_world/"
there are still references to using mimx8m tool chain and drivers but we will revisit
these later, I tried to make the change but gave up as it requires defining all of SOC
structure will rely on Ganesh to address these items.

atrivedi-tsavoritesi committed:
@FIR-13: Fix the lint and compliance warnings

atrivedi-tsavoritesi committed:
@FIR-13: Fixed compliance warning

atrivedi-tsavoritesi committed:
@FIR-13: Added tsi - Tsavorite Scalable Intelligence as a vendor

atrivedi-tsavoritesi committed:
@FIR-13: Fixed the sorting issue of vendors list

Commit:
removed unused symbol BOARD_TSI_MIMX8ML8_ADSP

Commit:
Synced for new target tsi/skyp/m85 that removed the ek_ prefix

Commit:
Removed white-spaces in main.c and Readme.rst

Commit:
renamed folder boards/tsi/tsi to boards/tsi/skyp

dineshReddy6381 commented:
Reviewed code and all review comments are fixed

Commit:
Removed compilance check warnings

dineshReddy6381 reviewed (approved):
reviewed code

atrivedi-tsavoritesi commented:
Approving the change
